src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -371,7 +381,67 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 			case ITermOscPt.SetMark: {
 				this._createOrGetCommandDetection(this._terminal).handleGenericCommand({ genericMarkProperties: { disableCommandStorage: true } });
 			}
+			default: {
+				// Checking for known `<key>=<value>` pairs.
+				const { key, value } = parseKeyValueAssignment(command);
+
+				if (value === undefined) {
+					// No '=' was found, so it's not a property assignment.
+					return true;
+				}
+
+				switch (key) {
+					case ITermOscPt.CurrentDir:
+						// Encountered: `OSC 1337 ; CurrentDir=<Cwd> ST`
+						this._updateCwd(value);
+						return true;
+				}
+			}
 		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	private _doHandleSetWindowsFriendlyCwd(data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command, ...args] = data.split(';');
+		switch (command) {
+			case '9':
+				// Encountered `OSC 9 ; 9 ; <cwd> ST`
+				if (args.length) {
+					this._updateCwd(args[0]);
+				}
+				return true;
+		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	/**
+	 * Handles the sequence: `OSC 7 ; scheme://cwd ST`
+	 */
+	private _doHandleSetCwd(data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command] = data.split(';');
+
+		// We need to manually make sure the given URI is not merely `file://` because The `URI.parse` handles it
+		// exactly as it handles `file:///` (which is a valid URI for us here).
+		if (command.startsWith('file://') && command.length > 7) {
```

Comment:
Is this a more correct condition to match on file:// followed by zero or more characters and then another `/`?

```
command.match(/^file:\/\/.*\//)
```